src/Sentry/SentryClientExtensions.cs

```diff
@@ -61,12 +61,49 @@ public static void CaptureUserFeedback(this ISentryClient client, SentryId event
         }
     }
 
+    /// <summary>
+    /// Flushes the queue of captured events until the timeout set in <see cref="SentryOptions.FlushTimeout"/>
+    /// is reached.
+    /// </summary>
+    /// <param name="client">The Sentry client.</param>
+    /// <remarks>
+    /// Blocks synchronously. Prefer <see cref="FlushAsync"/> in async code.
+    /// </remarks>
+    public static void Flush(this ISentryClient client) =>
+        client.FlushAsync().GetAwaiter().GetResult();
+
+    /// <summary>
+    /// Flushes the queue of captured events until the timeout is reached.
+    /// </summary>
+    /// <param name="client">The Sentry client.</param>
+    /// <param name="timeout">The amount of time allow for flushing.</param>
+    /// <remarks>
+    /// Blocks synchronously. Prefer <see cref="ISentryClient.FlushAsync(TimeSpan)"/> in async code.
+    /// </remarks>
+    public static void Flush(this ISentryClient client, TimeSpan timeout) =>
+        client.FlushAsync(timeout).GetAwaiter().GetResult();
+
+    /// <summary>
+    /// Flushes the queue of captured events until the timeout set in <see cref="SentryOptions.FlushTimeout"/>
+    /// is reached.
+    /// </summary>
+    /// <param name="client">The Sentry client.</param>
+    /// <returns>A task to await for the flush operation.</returns>
+    public static Task FlushAsync(this ISentryClient client)
+    {
+        var options = client.GetSentryOptions() ?? new SentryOptions();
```

Comment:
Is that the intended way? The new options would not have a DNS and I think that would get caught somewhere in the transport?

Reply:
That instance is only used temporarily, to retrieve the default timeout value in the rare case that the `GetSentryOptions` method returns null.  That would only happen if `ISentryClient` was something other than expected, such as a test substitute, and only if the `SentryOptionsForTestingOnly` property wasn't set.  So just a final check to get back the 2 second default without putting that in two places.